p2p/server.py

```diff
@@ -179,9 +180,13 @@ def _find_internal_ip_on_device_network(self, upnp_dev: upnpclient.upnp.Device)
         discover_timeout = 10 * REPLY_TIMEOUT
         # Use loop.run_in_executor() because upnpclient.discover() is blocking and may take a
         # while to complete.
-        devices = await self.wait_first(
-            loop.run_in_executor(None, upnpclient.discover),
-            timeout=discover_timeout)
+        devices: Iterable[upnpclient.Device] = []
+        try:
+            devices = await self.wait_first(
+                loop.run_in_executor(None, upnpclient.discover),
+                timeout=discover_timeout)
+        except TimeoutError:
+            self.logger.info("Timeout waiting for UPNP-enabled devices")
```

Comment:
You could return here and then avoid having to declare devices before the try/except